torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -60,6 +60,10 @@ std::map<at::ScalarType, ncclDataType_t> ncclDataType = {
     {at::kLong, ncclInt64},
     {at::kHalf, ncclHalf},
     {at::kBool, ncclUint8},
+    // TODO: need per collective handling
+    // (e.g., fp8 allgather OK, reduce-scatter NO)
+    {at::kFloat8_e5m2, ncclInt8},
```

Comment:
For provenance, @awgu said that these look fine

Reply:
Or my bad, I meant more that reviewing / figuring out the right way to do this should be fine. I am not too knowledgeable on ProcessGroupNCCL.

Let me cc: @H-Huang @kwen2501 @wconstab

Reply:
looks fine to me, for robustness we  will probably need to add validity checks in the reduce ops to ensure these dtypes arent being used, something like: https://github.com/pytorch/pytorch/blob/8162f4170b8c7df3bab503103544e1d7cbf8b052/torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp#L1404

either that or add another argument to `getNcclDataType` to pass in the collective to check there https://github.com/pytorch/pytorch/blob/8162f4170b8c7df3bab503103544e1d7cbf8b052/torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp#L69